sentry-ruby/lib/sentry/hub.rb

```diff
@@ -229,6 +229,34 @@ def with_session_tracking(&block)
       end_session
     end
 
+    def get_traceparent
+      return nil unless current_scope
+
+      current_scope.get_span&.to_sentry_trace ||
+        current_scope.propagation_context&.get_traceparent
+    end
+
+    def get_baggage
+      return nil unless current_scope
+
+      current_scope.get_span&.to_baggage ||
+        current_scope.propagation_context&.get_baggage&.serialize
```

Comment:
I think a scope's `propagation_context` should not be `nil`?

Also, `PropagationContext#get_baggage` should also not be `nil` as the early return in `populate_head_baggage` is not supposed to happen (explained in its comment).